### PR TITLE
Yet another react 0.1.4 upgrade PR

### LINKED ISSUE #89: React 14 update
Are there plans to update core and dependancies to use React 14?

### PR DESCRIPTION
**Pull request by thiagofelix**
Just submitting the work I had already started few days ago, also took the change to move the dependencies into peerDependencies so we don't have to force host projects into a specific react version.

Upgrade went smoothly as described by #95 already, there isn't much differences between both, but I believe this one has less changes.

There is still warnings with ReactCSSTransitionGroup ``Failed propType: transitionEnterTimeout wasn't supplied to ReactCSSTransitionGroup`` that couldn't be fixed because it breaks the animations/transitions.

The touchstone starter and tasks projects need also to be updated to work correctly after this change.

fixes #89 #95 

Thanks

### CODE REVIEW

**Comment by dcousens** [comment]
@thiagofelix where were your results with touchstonejs-starter? Work straight up also?

Awesome work!

**Comment by thiagofelix** [comment]
@dcousens changes on starter should land in another PR. 

Changes for this PR are probably going to be published on a newer version number ( 0.7.0 ) which leaves starter workin, until it upgrades as well.

But at least if you want to try it on react 0.14 right away you jave a running version.

Thanks

File: src/core/animation.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ Mixins.ScrollContainerToTop = {
 	},
 	scrollContainerToTop () {
 		if (!this.isMounted() || !this.refs.scrollContainer) return;
-		this._scrollContainerAnimation = scrollToTop(React.findDOMNode(this.refs.scrollContainer), {
+		this._scrollContainerAnimation = scrollToTop(ReactDOM.findNode(this.refs.scrollContainer), {
```
Comment (pke) [inline src/core/animation.js]: This `findNode` is [not even needed](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#dom-node-refs) in react 0.14 anymore. One could simply write `this.refs.scrollContainer`.

File: src/core/ViewManager.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 var blacklist = require('blacklist');
 var classNames = require('classnames');
 var ErrorView = require('./ErrorView');
-var React = require('react/addons');
-var Transition = React.addons.CSSTransitionGroup;
+var React = require('react');
+var ReactCSSTransitionGroup = require('react-addons-css-transition-group');
```
Comment (pke) [inline src/core/ViewManager.js]: Any reason why not only change that to `var Transition = require("react-addons-css-transition-group")` and have not change the rest of the file?

File: src/core/animation.js (hunk 1 above)
Comment (thiagofelix) [inline src/core/animation.js]: That's correct, I am going to change it.

Thanks!

File: src/core/ViewManager.js (hunk 2 above)
Comment (thiagofelix) [inline src/core/ViewManager.js]: The reason is to keep naming/concepts closer to react itself hence redeuce cognitive overhead when learning the code.

Comment (pke) [inline src/core/ViewManager.js]: Understood!

**Comment by JedWatson** [comment]
Thanks @thiagofelix - this looks great. Im going to merge and publish as `0.7.0` like you suggest.

I'll update touchstone-starter to include `react-dom` and stop bundling `react/addons` as well.

Any chance you're willing to take on updating touchstone-starter to the new version once this is released?

**Comment by thiagofelix** [comment]
Hi @JedWatson 

Thanks for merging.

I begin doing that but I went with some trouble because of browserify and extracting the react and the app bundle into individual files, so I stop looking.

I am super busy next days so I am afraid I can't look at that for a while =\.

**Comment by JedWatson** [comment]
No worries @thiagofelix - I've finished that now as well :)

**Pull request merged**